### PR TITLE
Enable Automatic Merge of approved Pull Requests

### PR DESCRIPTION
**Pull request by jamacku**
Add automation that will merge approved PRs with passed CI checks automatically.

It uses the following GitHub Actions:

* [gather-pull-request-metadata](https://github.com/redhat-plumbers-in-action/gather-pull-request-metadata) - to gather important data about PRs (used in following actions)
* [pull-request-validator](https://github.com/redhat-plumbers-in-action/pull-request-validator) - to check if PR is approved and has passed CI checks and more
* [auto-merge](https://github.com/redhat-plumbers-in-action/auto-merge) - to merge approved PRs
* [issue-commentator](https://github.com/redhat-plumbers-in-action/issue-commentator) - to comment on PR with status of PR and merge status

Here are some examples of use from the systemd source-git. Please notice the `Pull Request validation` and `Auto Merge` sections:

* https://github.com/redhat-plumbers/systemd-rhel9/pull/281#issuecomment-2165555171
* https://github.com/redhat-plumbers/systemd-rhel9/pull/277#issuecomment-2139901121
* https://github.com/jamacku/nginx-container/pull/5#issuecomment-2182743930

/cc @phracek

### CODE REVIEW

File: .github/workflows/auto-merge-on-demand.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        required: true
+        default: '0'
+
+permissions:
+  contents: read
+
+jobs:
+  # Get all open PRs
+  gather-pull-requests:
+    if: github.repository == 'sclorg/nginx-container'
```
Comment (phracek) [inline .github/workflows/auto-merge-on-demand.yml:19]: I would prefer to have something like this https://github.com/sclorg/nginx-container/blob/master/.github/workflows/build-and-push.yml#L10

File: .github/workflows/auto-merge-on-demand.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        uses: redhat-plumbers-in-action/gather-pull-request-metadata@v1
+        with:
+          pr-number: ${{ matrix.pr-number }}
+
+      - id: pull-request-validator
+        name: Pull Request Validator
+        uses: redhat-plumbers-in-action/pull-request-validator@v2
+        with:
+          pr-metadata: ${{ steps.metadata.outputs.metadata }}
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (phracek) [inline .github/workflows/auto-merge-on-demand.yml:78]: I guess, default can be like here: https://github.com/sclorg/testing-farm-as-github-action/blob/main/action.yml#L27 No needed to have it in secrets.

**Review by phracek — CHANGES_REQUESTED** [review]
After a brief look. Some comments.

File: .github/workflows/auto-merge-on-demand.yml (hunk 2 above)
Comment (jamacku) [reply .github/workflows/auto-merge-on-demand.yml:78]: The pull request validator doesn't provide a default for the token input. It has to be explicitly passed in the workflow.

https://github.com/redhat-plumbers-in-action/pull-request-validator/blob/564abe63f0f453bdde54171fa575daf199913786/action.yml#L28-L30

**Pull request merged**